src/util/readUtf8FileWithBom.ts

```diff
@@ -0,0 +1,15 @@
+// ----------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// ----------------------------------------------------------------------------
+
+import * as fse from 'fs-extra';
+
+const utf8Bom = 65279;
+
+export async function readUtf8FileWithBom(fsPath: string): Promise<string> {
+    let contents = (await fse.readFile(fsPath, { encoding: "utf8" }));
+    if (contents.charCodeAt(0) === utf8Bom) {
+        contents = contents.slice(1);
```

Comment:
>contents.slice(1) [](start = 19, length = 17)

Will this end up re-allocating the entire string/file contents? If yes, maybe would be more efficient to read the first char only first, and then read the rest of the file (starting at char 1 if char 0 is BOM?) Might possibly make a difference if someone is re-opening a workspace with a number of large templates previously opened.

Reply:
It's a good suggestion, but complicates the read logic, and the BOM should be a rare case (only a couple of users have hit this). And there's always GC.